source/geometries/Next100.cc

```diff
@@ -154,13 +154,15 @@ namespace nexus {
 
     G4ThreeVector gate_pos(0., 0., -gate_zpos_in_vessel_);
     if (lab_walls_){
-      G4ThreeVector castle_pos(0., hallA_walls_->GetLSCHallACastleY(),
+      G4ThreeVector castle_pos(0., hallA_walls_->GetLSCHallACastleY() + shielding_->GetHeight(),
```

Comment:
What's the reason of adding the height of the castle?

Reply:
To see this you can assume that `hallA_walls_->GetLSCHallACastleY()` is zero, then you want the floor of the lead-castle to be aligned with the floor of the lab. In fact, instead of being the height it should be height/2.
<img width="532" alt="castle_floor" src="https://user-images.githubusercontent.com/35993868/189070329-ed357f3b-79e8-45e9-864f-0d109d15935b.png">